backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -172,23 +157,104 @@ public List<FileResponse> getAllFilesWithParent(UUID parentFolderId) {
     return asFileResponseList(allFiles);
   }
 
-  private List<FileResponse> asFileResponseList(List<FileMetadataEntity> allFiles) {
-    return allFiles.stream()
-        .map(folder -> modelMapper.map(folder, FileResponse.class))
-        .collect(Collectors.toList());
+  @Override
+  @Transactional(rollbackFor = Exception.class)
+  public FileResponse editFileContent(MultipartFile multipartFile, UUID fileId) {
+
+    FileMetadataEntity fileMetadata =
+        fileMetadataRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
+
+    checkFileTypeNotChanged(fileMetadata, multipartFile.getContentType());
+
+    fileMetadata.setType(multipartFile.getContentType());
```

Comment:
Since you already have a check about the file type not changed, do you actually need this here?

Reply:
the check only checks whether the old file metadata's type is the same as the multipart file's type
then I set the new type

it's dumb, I'll change it into something readable

Reply:
should be easier to read now